pkg/apis/build/v1alpha1/buildrun_types.go

```diff
@@ -23,7 +23,9 @@ const (
 // BuildRunSpec defines the desired state of BuildRun
 type BuildRunSpec struct {
 	// BuildRef refers to the Build
-	BuildRef BuildRef `json:"buildRef"`
+	//
+	// +optional
+	BuildRef *BuildRef `json:"buildRef"`
```

Comment:
No, this is fine. With the introduction of `BuildSpec`, the `BuildRef` needs to be optional now. Otherwise we could have both a build reference and an embedded build(spec).